pkg/apis/vpc/v1alpha1/vpcnetworkconfiguration_types.go

```diff
@@ -44,6 +44,12 @@ type VPCNetworkConfigurationSpec struct {
 type SharedSubnet struct {
 	// NSX path of Subnets created outside of the Supervisor to be associated with this vSphere Namespace
 	Path string `json:"path"`
+	// Indicates if this subnet is the default subnet for pods.
+	PodDefault bool `json:"podDefault,omitempty"`
+	// Indicates if this subnet is the default subnet for VMs.
+	VMDefault bool `json:"vmDefault,omitempty"`
+	// Name of the Subnet. If the name is empty, it will be derived from the shared Subnet path.
```

Comment:
/**
       * Name of Subnet for this shared Subnet. After creation, the name cannot be updated.
       * Name should follow the k8s standards <a href="https://kubernetes.io/docs/concepts/overview/working-with-objects/names/#dns-subdomain-names">DNS Subdomain Names</a>. 
       *
       * @field.optional If {@term unset}, the Subnet name will be the shared Subnet ID, and if there is conflict, the Subnet name will be the shared Subnet ID suffixed by 5 alphanumeric characters randomly generated.

Above is API doc I added in vmodl.
Could we make the “name" field immutable?